addons/survey/models/survey_survey.py

```diff
@@ -321,8 +314,17 @@ def copy_data(self, default=None):
     def toggle_active(self):
         super(Survey, self).toggle_active()
         activated = self.filtered(lambda survey: survey.active)
+        archived = self - activated
         activated.mapped('certification_badge_id').action_unarchive()
-        (self - activated).mapped('certification_badge_id').action_archive()
+        archived.mapped('certification_badge_id').action_archive()
+        if archived:
+            self.env['bus.bus'].sendone(
```

Comment:
Why doing that ? Is it really interesting ? I mean we don't do that for other documents ...

Reply:
Well I checked for the other objects (lead, slide course etc), it only notifies for the survey so thought it's handled by framework. I'll add the condition any way.